src/state_machine.zig

```diff
@@ -692,34 +692,59 @@ pub fn StateMachineType(comptime Storage: type, comptime constants_: struct {
             // If the transfer already exists, then it must not influence the overflow or limit checks.
             if (self.get_transfer(t.id)) |e| return create_transfer_exists(t, e);
 
+            const amount = amount: {
+                var amount: u64 = t.amount;
+                if (t.flags.debits_at_most) {
+                    const dr_limit = if (dr.flags.debits_must_not_exceed_credits)
+                        dr.credits_posted
+                    else
+                        std.math.maxInt(u64);
+                    const dr_available = dr_limit - dr.debits_posted - dr.debits_pending;
```

Comment:
It seems surprising to have `dr_available` be the potential diff between `maxInt(u64)` and the debit balance. As an interim variable it's not meaningful.

Can we rework this calculation to rather cache `dr_balance = dr.debits_posted + dr.debits_pending` (which would be more meaningful by itself), and then do the same for `cr_balance` and then put these together?

Reply:
Names like `debit balance` and `net dr/cr balance` are accounting terms—it would be great to phrase the calculation in those terms.

Reply:
Maybe I'm misunderstanding the definitions of "debit balance" I've found, but wouldn't it be `debits_posted + dr.debits_pending - credits_posted`, not `debits_posted + debits_pending`?

Reply:
The *debit balance* is only the sum of the debit side.

The *net debit balance* would be the sum of the debit side less the credit side, assuming the debit side is greater.

Reply:
Sometimes people say that an account has a *debit balance*, implying *net debit balance*.

Reply:
> The debit balance is only the sum of the debit side.

:+1: 

> Sometimes people say that an account has a debit balance, implying net debit balance.

Yeah, that is what confused me.